Remove unused 'backend_info'

okurz opened the pull request:
The column "backend_info" seems to have been prepared for use by
backends but I also could not find any reference in os-autoinst so this
commit will remove the table in the database, will remove references and
migrate databases by simply removing the column without replacement.

okurz commented:
Thanks to @r-richardson who pointed out that this might not be used. Added the `not-ready` label to keep it open for some weeks for consideration before accepting.

kraih commented:
The patch should probably remove [this line](https://github.com/os-autoinst/openQA/blob/c9f8b107d3c0b21a4ba8098aaf6d2fe95d0aeba3/lib/OpenQA/Schema/Result/Jobs.pm#L1442) too.

okurz commented:
> The patch should probably remove [this line](https://github.com/os-autoinst/openQA/blob/c9f8b107d3c0b21a4ba8098aaf6d2fe95d0aeba3/lib/OpenQA/Schema/Result/Jobs.pm#L1442) too.

Well, we still need "backend", I removed too much initially. I updated the code to not read/write backend_info only but still keep backend

Martchus commented on lib/OpenQA/Schema/Result/Jobs.pm:
```diff
@@ -1040,7 +1035,6 @@ sub update_backend ($self, $backend_info) {
     $self->update(
         {
             backend => $backend_info->{backend},
-            backend_info => encode_json($backend_info->{backend_info})});
```
You still need the closing braces and semicolon.

okurz replied:
Sorry, with stupid mistakes like those I really feel I am not yet 100% healthy again

kalikiana reviewed (approved):
Nice find

kalikiana commented:
@mergifyio rebase
